activesupport/lib/active_support/inflector/transliterate.rb

```diff
@@ -75,13 +75,21 @@ def parameterize(string, sep = '-')
       # Turn unwanted chars into the separator
       parameterized_string.gsub!(/[^a-z0-9\-_]+/i, sep)
       unless sep.nil? || sep.empty?
-        re_sep = Regexp.escape(sep)
+        if sep == "-".freeze
```

Comment:
@schneems the "sep" variable in the definition of the parameterize function is not using the .freeze when sep is not present in the call. Will this if catch or miss the case when somebody call parameterize("string")?
If the comparison takes in consideration the string (and not the object id), this probably won't be a problem... Does it make sense?
I noticed you used the sep = "-".freeze in the parameterize function on file 'activesupport/lib/active_support/core_ext/string/inflections.rb'